test/OpenTelemetry.Instrumentation.Http.Tests/HttpWebRequestTests.Basic.netfx.cs

```diff
@@ -95,6 +95,43 @@ public async Task HttpWebRequestInstrumentationInjectsHeadersAsync()
             parent.Stop();
         }
 
+        [Fact]
+        public async Task HttpWebRequestInstrumentationInjectsHeadersAsyncWhenActivityIsNotRecorded()
+        {
+            var activityProcessor = new Mock<BaseProcessor<Activity>>();
+            using var shutdownSignal = Sdk.CreateTracerProviderBuilder()
+                .AddProcessor(activityProcessor.Object)
+                .AddHttpWebRequestInstrumentation()
+                .Build();
+
+            var request = (HttpWebRequest)WebRequest.Create(this.url);
+
+            request.Method = "GET";
+
+            var parent = new Activity("parent")
+                .SetIdFormat(ActivityIdFormat.W3C)
+                .Start();
+            parent.TraceStateString = "k1=v1,k2=v2";
+            parent.ActivityTraceFlags = ActivityTraceFlags.None;
+
+            using var response = await request.GetResponseAsync();
+
+            Assert.Empty(activityProcessor.Invocations);
+            var activity = Activity.Current;
+
+            Assert.Equal(parent.TraceId, activity.Context.TraceId);
```

Comment:
i think what you want to do is to register a customPropagator, and inside that validate that ActivityContext is the one from parent

Reply:
just updated. i will might have to change after this or after the other pr is merged.